src/fsverity/mod.rs

```diff
@@ -19,3 +25,61 @@ impl FsVerityHashValue for Sha512HashValue {
     const ALGORITHM: u8 = 2;
     const EMPTY: Self = [0; 64];
 }
+
+/// A verity comparison failed.
+#[derive(Error, Debug)]
+pub enum CompareVerityError {
+    #[error("failed to read verity")]
+    Measure(#[from] MeasureVerityError),
+    #[error("fsverity is not enabled on target file")]
+    VerityMissing,
+    #[error("Expected digest {expected} but found {found}")]
+    DigestMismatch { expected: String, found: String },
+}
+
+/// Require the fsverity digest to be present.
+pub fn measure_verity_digest<F: AsFd, H: FsVerityHashValue>(
+    fd: F,
+) -> Result<H, CompareVerityError> {
+    match ioctl::fs_ioc_measure_verity::<_, H>(fd)? {
+        Some(found) => Ok(found),
+        None => Err(CompareVerityError::VerityMissing),
```

Comment:
It feels kinda weird how we take a syscall errno, convert it to None, then convert it back to a synthetic error.  But I think that's my C brain talking.

In reality `VerityMissing` is a way more meaningful error than `ENODATA` or `ENOTTY` or whatever random crap might be thrown at us.